lib/facter/facts/solaris/virtual.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Facts
+  module Solaris
+    class Virtual
+      FACT_NAME = 'virtual'
+
+      def initialize
+        @log = Facter::Log.new(self)
+      end
+
+      def call_the_resolver
+        @log.debug('Solaris Virtual Resolver')
+
+        fact_value = check_ldom || check_zone || 'physical'
+
+        @log.debug("Fact value is: #{fact_value}")
+
+        Facter::ResolvedFact.new(FACT_NAME, fact_value)
+      end
+
+      def check_ldom
+        @log.debug('Checking LDoms')
+        Facter::Resolvers::Solaris::Ldom.resolve(:role_impl)
+      end
+
+      def check_zone
+        @log.debug('Checking LDoms')
+        zone_name = Facter::Resolvers::SolarisZoneName.resolve(:current_zone_name)
+
+        return if zone_name == 'global'
+
+        zone_name
```

Comment:
https://github.com/puppetlabs/facter/blob/fd44acffcc64544da913e49acfa02bf0da7be62c/lib/src/facts/solaris/virtualization_resolver.cc#L32 vm::zone is 'zone' => https://github.com/puppetlabs/facter/blob/4844068e8dfc5118258a42cd73327443a07778ca/lib/inc/facter/facts/vm.hpp#L139

Reply:
Updated, thanks!